charts/sftpgo/templates/deployment.yaml

```diff
@@ -38,6 +38,13 @@ spec:
             - sftpgo
             - serve
           env:
+            {{- range .Values.envFromSecrets }}
+            - name: {{ .name | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .valueFrom.secretKeyRef.name | quote }}
+                  key: {{ .valueFrom.secretKeyRef.key | quote }}
+            {{- end }}
```

Comment:
Let's move this to the very end of the `env` section so it comes as last in the order of precedence. Also, use `toYaml` and `with` instead to make it more generic (support configmaps, etc).

Reply:
Ok good point on making it more generic @sagikazarmark , will do that. But i would still suggest to leave it at the top of `env` section to allow clients to use envs set from secrets or configmaps in key value envs. Something like this

```
env:
  SFTPGO_DATA_PROVIDER__CONNECTION_STRING: "postgres://$(DB_USER):$(DB_PASSWORD)@db-sftpgo:5432/sftpgo"

envVars:
  - name: DB_PASSWORD
    valueFrom:
      secretKeyRef:
        key: passwd
        name: postgres-owner-sftpgo
  - name: DB_USER
    valueFrom:
      secretKeyRef:
        key: username
        name: postgres-owner-sftpgo
```

Reply:
made the needed changes please review

Reply:
I see. Well, even then, I'd move the envVars section to after the manually defined env vars as those shouldn't be overwritten.